src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala

```diff
@@ -21,7 +21,7 @@ object DictFunctions extends RegistryFunctions {
       LowerBoundOnOrderedCollection(dict, key, onKey=true),
       If(IsNA(dict),
         NA(default.typ),
-        If(ApplyComparisonOp(EQWithNA(key.typ), GetField(ArrayRef(ToArray(dict), i), "key"), key),
+        If(ApplyComparisonOp(EQ(key.typ), GetField(ArrayRef(ToArray(dict), i), "key"), key),
```

Comment:
So this will always return missing if the key is missing (rather than the default)?

Reply:
yep. this is the current behavior in AST as well.